src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -217,7 +213,7 @@ private NetworkPeerConnectionParameters CreateNetworkPeerConnectionParameters()
         /// <returns>When criteria is met returns <c>true</c>, to allow connection.</returns>
         private bool AllowClientConnection(TcpClient tcpClient)
         {
-            if (this.networkPeerDisposer.ConnectedPeersCount >= MaxConnectionThreshold)
+            if (this.networkPeerDisposer.ConnectedPeersCount >= this.connectionManagerSettings.MaxConnections)
```

Comment:
This logic wasn't right.
In StratisX we can see the following:
https://github.com/stratisproject/stratisX/blob/f9ef7fbb76a385b207c92c3de1fecb31047e441a/src/net.cpp#L820-L845

So I think this should be changed to 
` if (this.networkPeerDisposer.ConnectedPeers.Count(p => p.Value.Inbound) >= this.connectionManagerSettings.MaxOutboundConnections - this.connectionManagerSettings.MaxOutboundConnections)`

Also you'd need to make connectedPeers public get 
```
/// <summary>Mapping of connected peers by their connection ID.</summary>
public ConcurrentDictionary<int, INetworkPeer> ConnectedPeers { get; }
```